src/runtime/manifest.ts

```diff
@@ -184,6 +181,11 @@ export class Manifest {
     }
     return this._id;
   }
+
+  get idGenerator() {
+    return this._idGenerator;
```

Comment:
with this, can _idGenerator be private?

Reply:
renamed it idGenerator and removed the getter, here and in arc.ts. It's readonly already; private with a getter is basically the same as readonly :)